core/src/main/java/org/infinispan/statetransfer/CommitManager.java

```diff
@@ -109,21 +118,29 @@ public final void stopTrack(Flag track) {
          return CompletableFutures.completedNull();
       }
       ByRef<CompletionStage<Void>> byRef = new ByRef<>(null);
-      tracker.compute(entry.getKey(), (o, discardPolicy) -> {
-         if (discardPolicy != null && discardPolicy.ignore(operation)) {
+      tracker.compute(segment, (s, entries) -> {
+         if (entries == null) {
+            entries = new ConcurrentHashMap<>();
+         }
+
+         entries.compute(entry.getKey(), (o, discardPolicy) -> {
```

Comment:
I would return the value from the tracker compute, so we don't need to indent this. Is a bit cleaner imo.